From 16bf951d16ff45db1c6094a3793c26022b674a62 Mon Sep 17 00:00:00 2001
From: Steve Best <sbest@redhat.com>
Date: Wed, 27 Aug 2014 20:55:24 -0400
Subject: [usb] xhci: convert TRB_CYCLE to le32 before using it to set Link TRB's cycle bit

Message-id: <20140827205523.3990.37429.sendpatchset@ibm-p7r2-01.lab.bos.redhat.com>
Patchwork-id: 89287
O-Subject: [PATCH RHEL7.1 BZ1123119 2/2] xhci: convert TRB_CYCLE to le32 before using it to set Link TRB's cycle bit
Bugzilla: 1123119
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

RHBZ#:
------
https://bugzilla.redhat.com/show_bug.cgi?id=1123119

Description:
------------
This patch converts TRB_CYCLE to le32 to update correctly the Cycle Bit in
'control' field of the link TRB.
This bug was found using sparse.

Signed-off-by: Xenia Ragiadakou <burzalodowa@gmail.com>
Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>

RHEL Version Found:
-------------------
7.0

Upstream Status:
----------------
http://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=587194873820a4a1b2eda260ac851394095afd77

----------------
Steve Best

Proposed Patch:
---------------

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-mem.c b/drivers/usb/host/xhci-mem.c
index 83bcd13..6f0c7f4 100644
--- a/drivers/usb/host/xhci-mem.c
+++ b/drivers/usb/host/xhci-mem.c
@@ -57,7 +57,7 @@ static struct xhci_segment *xhci_segment_alloc(struct xhci_hcd *xhci,
  /* If the cycle state is 0, set the cycle bit to 1 for all the TRBs */
  if (cycle_state == 0) {
   for (i = 0; i < TRBS_PER_SEGMENT; i++)
-   seg->trbs[i].link.control |= TRB_CYCLE;
+   seg->trbs[i].link.control |= cpu_to_le32(TRB_CYCLE);
  }
  seg->dma = dma;
  seg->next = NULL;
@@ -308,7 +308,8 @@ static void xhci_reinit_cached_ring(struct xhci_hcd *xhci,
     sizeof(union xhci_trb)*TRBS_PER_SEGMENT);
   if (cycle_state == 0) {
    for (i = 0; i < TRBS_PER_SEGMENT; i++)
-    seg->trbs[i].link.control |= TRB_CYCLE;
+    seg->trbs[i].link.control |=
+     cpu_to_le32(TRB_CYCLE);
   }
   /* All endpoint rings have link TRBs */
   xhci_link_segments(xhci, seg, seg->next, type);
-- 
1.7.1